PMSCore/User.cs

```diff
@@ -0,0 +1,247 @@
+namespace PMSCore;
+
+public abstract class User : IUser
+{
+    protected User(string userId, HashSet<Permission> userPermissions)
+    {
+        UserId = userId;
+        UserPermissions = userPermissions;
+        UserReport = string.Empty;
+    }
+
+    private void AppendToReport(string message)
+    {
+        Console.WriteLine(message);
+        UserReport += $"[{DateTime.UtcNow}] [{UserId}] [{message}]\n";
```

Comment:
Console.WriteLine(message);
UserReport += $"[{DateTime.UtcNow}] [{UserId}] [{message}]\n";

maybe sth like: 

UserReport += $"[{DateTime.UtcNow}] [{UserId}] [{message}]\n";
_logger.LogInformation(message);

so it's not logging directly to AppendToReport

Reply:
User like the entities contains no logger only the managers do, otherwise we gonna end up having logger file per user created not really worth in my opinion what do you think?

Reply:
I agree, having loggers for each user can make things more complicated.
Keeping logging centralised in managers like StateManager is a more simple approach.